nemo/collections/asr/parts/submodules/cuda_graph_rnnt_greedy_decoding.py

```diff
@@ -0,0 +1,427 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import contextlib
+
+import numpy as np
+import torch
+
+try:
+    from cuda import cuda, cudart, nvrtc
+
+    HAVE_CUDA_PYTHON = True
+except ImportError:
+    HAVE_CUDA_PYTHON = False
+from typing import List, Optional
+
+from nemo.collections.asr.parts.utils import rnnt_utils
+from nemo.core.classes import Typing, typecheck
+from nemo.core.neural_types import AcousticEncodedRepresentation, ElementType, HypothesisType, LengthsType, NeuralType
+from nemo.core.utils.cuda_python_utils import (
+    assert_drv,
+    check_cuda_python_cuda_graphs_conditional_nodes_supported,
+    cu_call,
+)
+from nemo.utils import logging
+
+
+def run_nvrtc(kernel_string, kernel_name):
+    err, prog = nvrtc.nvrtcCreateProgram(str.encode(kernel_string), b"while_loop_conditional.cu", 0, [], [])
```

Comment:
looks like this could be made generic

```suggestion
    err, prog = nvrtc.nvrtcCreateProgram(str.encode(kernel_string), str.encode(f"{kernel_name}.cu"), 0, [], [])
```

Reply:
The trouble with that suggestion is that it assumes a file contains only a single kernel inside. I actually already broke this assumption when I made two "timer kernels". One to record the time, and one to print out the time different from the recorded time. This also required a global `__device__` int64_t array of times as well. That code is not committed, since I used it just for debugging. But the point is that it is not necessarily correct that we can always assume one kernel per file. So I don't want to modify this code right now. The actual "file name" doesn't matter anyway.